src/profile/webid.ts

```diff
@@ -195,3 +195,16 @@ export function getPodUrlAllFrom(
   );
   return Array.from(result);
 }
+
+/**
+ * Get the WebID Profile document dataset.
+ *
+ * @param webId The WebID of the agent whose WebID Profile dataset is to be fetched.
+ * @returns a SolidDataset for the WebID Profile document.
+ * @since unreleased
+ */
+export async function getWebIdDataset(
+  webId: WebId
+): Promise<SolidDataset | null> {
```

Comment:
```suggestion
): Promise<SolidDataset> {
```

Reply:
You may alternatively wish to use 
```suggestion
): Promise<ReturnType<getSolidDataset>> {
```

As `getSolidDataset` returns a `SolidDataset & WithResourceInfo`

Reply:
In this case I think it's better to only treat this as a a `SolidDataset`, as that's already a misnomer, but the `WithResourceInfo` type is really relevant only for Solid resources, which isn't an assumption we can make about WebID profiles.

Reply:
Actually my bad, you're right: since other functions of this module (e.g. `getAltProfileUrlAllFrom`) expect a `SolidDataset & WithResourceInfo`, it's probably better to remain aligned.